core/store/src/db.rs

```diff
@@ -230,6 +230,11 @@ pub trait Database: Sync + Send {
 
     /// Create checkpoint in provided path
     fn create_checkpoint(&self, path: &std::path::Path) -> anyhow::Result<()>;
+
+    /// Returns database to which updates should be written, if it is not `self`.
+    /// Only meaningful for split db and cold db.
+    /// Otherwise just returns None and should be interpreted as `Arc<self>`.
```

Comment:
Can we just drop the Option and return Self in those cases? It would be easier to use.